### PR TITLE
Add script for profiling startup using flutter trace on Android

### PR DESCRIPTION
**Pull request by eseidelGoogle**
This will work after the next Engine roll.  Currently it requires
a fix to tracing which isn't in the released engine.

Next steps are to make this work on a bot, and to add iOS support.

@abarth @chinmaygarde

### CODE REVIEW

File: dev/profile_startup.dart [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    if (checked)
+      throw errorDescription;
+  }
+  if (results.stdout.trim().isNotEmpty)
+    print(results.stdout.trim());
+  return results.stdout;
+}
+
+double timeToFirstFrame(trace) {
+  // FIXME: Sort! Trace events are not guarenteed to be in timestamp order.
```
Comment (Hixie) [inline dev/profile_startup.dart]: use TODO() rather than FIXME

File: dev/profile_startup.dart [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  int firstTimeStamp = events[0]['ts'].toInt();
+  var firstSwap = events.firstWhere((e) => e['name'] == 'NativeViewGLSurfaceEGL:RealSwapBuffers');
+  int swapStart = firstSwap['ts'].toInt();
+  int swapEnd = swapStart + firstSwap['dur'].toInt();
+  return (swapEnd - firstTimeStamp) / 1000; // microseconds to milliseconds.
+}
+
+Future<double> test(String tracesDir, String projectPath, int runNumber) async {
+  // If we used package:path we could grab the basename of project_path
+  // and include that in the trace_name.
```
Comment (Hixie) [inline dev/profile_startup.dart]: if we get to that point we should just make this a feature of the `flutter` tool

**Comment by eseidelGoogle** [comment]
I could make this a flutter command instead.  In many ways that might be easier.

**Comment by Hixie** [comment]
LGTM if travis goes green

**Comment by Hixie** [comment]
actually I don't know if travis even checks this directory. Try running "flutter analyze profile_startup.dart" in this directory.

**Comment by chinmaygarde** [comment]
Since the iOS target is launched via Xcode, I can add another Xcode scheme that specifically adds the flag

**Pull request merged**